test/integration/bucketfs_test.py

```diff
@@ -0,0 +1,170 @@
+            {
+                "EXAClusterOS/ScriptLanguages-standard-EXASOL-7.1.0-slc-v4.0.0-CM4RWW6R.tar.gz",
+            },
+        ),
+        (
+            "myudfs",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            set(),
+        ),
+        (
+            "jdbc_adapter",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            set(),
+        ),
+    ],
+)
+def test_list_buckets(bucket, url, username, password, expected):
+    service = Bucket(bucket, url, username, password)
+    actual = {bucket for bucket in service}
+    assert expected.issubset(actual)
+
+
+@pytest.mark.parametrize(
+    "bucket,url,username,password,data",
+    [
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            bytes([65, 66, 67, 68, 69, 70]),
+        ),
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            [b"12", b"34", b"56", b"78"],
+        ),
+        (
+            "default",
+            "http://127.0.0.1:6666",
+            "w",
+            "write",
+            (b"1" for _ in range(0, 10)),
+        ),
+    ],
+)
+def test_upload_to_bucket(bucket, url, username, password, data):
+    upload_file = "Uploaded-File-{rnd}.bin".format(
+        rnd="".join(random.choice(string.hexdigits) for _ in range(0, 10))
+    )
+    bucket = Bucket(bucket, url, username, password)
+
+    # make sure this file does not exist yet in the bucket
+    assert upload_file not in bucket.files
+
+    # run test scenario
+    bucket.upload(upload_file, data)
+
+    # assert expectations
+    assert upload_file in bucket.files
+
+    # cleanup
+    _, _ = _delete_file(url, bucket.name, username, password, upload_file)
+
+
+@dataclass
+class File:
+    path: str
+    content: bytes
+
+
+@pytest.mark.parametrize(
+    "service,bucket,username,password,file",
+    [
+        (
+            "http://127.0.0.1:6666",
+            "default",
+            "w",
+            "write",
+            File(path="hello.txt", content=b"foobar"),
+        ),
+        (
+            "http://127.0.0.1:6666",
+            "default",
+            "w",
+            "write",
+            File(path="foo/bar/hello.bin", content=b"foobar"),
+        ),
+    ],
+)
+def test_download_file_from_bucket(service, bucket, username, password, file):
+    uploaded_file, _ = _upload_file(
+        service, bucket, username, password, file.path, file.content
+    )
+    bucket = Bucket(bucket, service, username, password)
+
+    # make sure this file does not exist yet in the bucket
+    assert uploaded_file in bucket.files
+
+    # run test scenario
+    data = as_bytes(bucket.download(uploaded_file))
+
+    # assert expectations
+    expected = file.content
+    assert data == expected
+
+    # clean up
+    _, _ = _delete_file(service, bucket.name, username, password, file.path)
```

Comment:
Try finally around the whole test or function scope fixture, which provides the name and cleans it up